### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by AdamCraftmaster**
This configures the logging system and allows for both configurations from the deploy directory or from a USB located on /media/sda1.

### CODE REVIEW

**Review by arfrie22 — APPROVED** [review]

File: src/main/java/frc/robot/logging/Configs.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+import org.apache.logging.log4j.Logger;
+
+public class Configs {
+  private static String config = "";
+  // possible locations for the config
+  public static String[] configLocations = { "./csv.txt", "./csv.txt" };
```
Comment (john-petrangelo) [inline src/main/java/frc/robot/logging/Configs.java:13]: Isn't this the same location twice?

File: src/main/java/frc/robot/logging/Configs.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        break;
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+    config = contentBuilder.toString();
+    LOGGER.info(config);
+  }
+}
```
Comment (john-petrangelo) [inline src/main/java/frc/robot/logging/Configs.java:37]: Missing final newline.

File: src/main/java/frc/robot/logging/RobotLogManager.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+  private static boolean initialized = false;
+
+  private static File directory;
+
+  private static String[] filepaths = { //File paths go in this array
+     "/media/sda1/logging/log4j2.yaml",
+    // "/media/sda2/logging/log4j2.yaml",
+      "/media/sda1/logging/log4j2-test.yaml",
+    // "/media/sda2/logging/log4j2-test.yaml",
```
Comment (john-petrangelo) [inline src/main/java/frc/robot/logging/RobotLogManager.java:25]: Is there a reason to keep these commented out lines?

**Review by john-petrangelo — APPROVED** [review]
Looks good to me, I had a couple of questions in there.

I assume you've tested that:
- it works as expected (i.e. it does what the config file tells it to do)
- if the config file is borked the program doesn't crash, it simply runs without logging
- if there is no USB drive plugged in, it simply runs without logging

**Comment by AdamCraftmaster** [comment]
> * if there is no USB drive plugged in, it simply runs without logging

Currently when there is no USB, it should use a deploy config that saves it to the root of the roborio

**Comment by arfrie22** [comment]
> > * if there is no USB drive plugged in, it simply runs without logging
> 
> Currently when there is no USB, it should use a deploy config that saves it to the root of the roborio

We should not be saving to the root because the roborio has very little storage. When we had it saving to root we ended up crashing the rio due to running out of space. It should just not save logs if no USB is plugged in and throw an warning in the console.

File: src/main/java/frc/robot/logging/RobotLogManager.java (hunk 3 above)
Comment (arfrie22) [reply src/main/java/frc/robot/logging/RobotLogManager.java:25]: We are not sure what port the usb will be plugged into

**Pull request merged**